### PR TITLE
Implement update module

### PR DESCRIPTION
**Pull request by RashadAnsari**

### CODE REVIEW

File: ghasedak-test/src/test/scala/im/ghasedak/server/utils/UpdateMatcher.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      override def onError(t: Throwable): Unit = {}
+
+      override def onCompleted(): Unit = {}
+
+      override def onNext(value: ResponseGetDifference): Unit = {
+        hashAnyUpdate = true
+      }
+    }
+    stub.getDifference(RequestGetDifference(Some(seqState)), streamObserver)
+    Thread.sleep(sleepForUpdates)
```
Comment (amsjavan) [inline ghasedak-test/src/test/scala/im/ghasedak/server/utils/UpdateMatcher.scala]: What is this?

File: ghasedak-rpc/src/main/scala/im/ghasedak/server/rpc/update/UpdateServiceHelper.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    seqState:         ApiSeqState,
+    responseObserver: StreamObserver[ResponseGetDifference]): Unit = {
+    val messageId = seqUpdateExt.getMessageId(seqState)
+    val pulsarSeq = MessageId.toJava(messageId).asInstanceOf[MessageIdImpl].getEntryId
+    if (seqState.seq == pulsarSeq) {
+      val reader = seqUpdateExt.getDifferenceReader(userId, tokenId, messageId)
+      getDifference(userId, tokenId, reader, responseObserver)
+    } else {
+      log.error(
+        "Unmatch pulsar and client sequence, userId: {}, tokenId: {}, userSeq: {}, pulsarSeq: {}",
```
[the remark this answers is not in the document]
Comment (RashadAnsari) [reply ghasedak-rpc/src/main/scala/im/ghasedak/server/rpc/update/UpdateServiceHelper.scala]: Removed

File: ghasedak-rpc/src/main/scala/im/ghasedak/server/rpc/update/UpdateServiceHelper.scala [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    tokenId:          String,
+    reader:           Reader[UpdateMapping],
+    responseObserver: StreamObserver[ResponseGetDifference],
+    consumerMessage:  ConsumerMessage[UpdateMapping]): Unit = {
+    val response = buildDifference(tokenId, consumerMessage)
+    try {
+      responseObserver.onNext(response)
+      getDifference(userId, tokenId, reader, responseObserver)
+    } catch {
+      case _: Throwable ⇒ reader.closeAsync
```
[the remark this answers is not in the document]
Comment (RashadAnsari) [reply ghasedak-rpc/src/main/scala/im/ghasedak/server/rpc/update/UpdateServiceHelper.scala]: Removed

File: ghasedak-rpc/src/main/scala/im/ghasedak/server/rpc/update/UpdateServiceImpl.scala [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  // todo: use separate dispatcher for rpc handlers
+  override implicit val ec: ExecutionContext = system.dispatcher
+
+  override val db: PostgresProfile.backend.Database = DbExtension(system).db
+
+  override val log: LoggingAdapter = Logging.getLogger(system, this)
+
+  protected val seqUpdateExt = SeqUpdateExtension(system)
+
+  implicit private def onFailure: PartialFunction[Throwable, RpcError] = {
```
[the remark this answers is not in the document]
Comment (RashadAnsari) [reply ghasedak-rpc/src/main/scala/im/ghasedak/server/rpc/update/UpdateServiceImpl.scala]: Done

File: ghasedak-update/src/main/scala/im/ghasedak/server/DeliveryOperations.scala [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    } yield seqState
+  }
+
+  private def deliverUpdate(
+    userId:    Int,
+    mapping:   UpdateMapping,
+    reduceKey: Option[String] = None): Future[ApiSeqState] = {
+    val topic = getUserUpdateTopic(userId)
+    val producerConfig = getBaseUserUpdateProducerConfig.copy(topic = topic)
+    val producer = pulsarClient.producer[UpdateMapping](producerConfig)
```
[the remark this answers is not in the document]
Comment (RashadAnsari) [reply ghasedak-update/src/main/scala/im/ghasedak/server/DeliveryOperations.scala]: We will fix it in future

File: ghasedak-test/src/test/scala/im/ghasedak/server/utils/UpdateMatcher.scala (hunk 1 above)
Comment (RashadAnsari) [reply ghasedak-test/src/test/scala/im/ghasedak/server/utils/UpdateMatcher.scala]: I will remove it after solving tests problem

File: project/Dependencies.scala [hunk 6]
```diff
@@ -42,7 +43,13 @@ object Dependencies {
     val pulsar4s = Seq(
       "com.sksamuel.pulsar4s" %% "pulsar4s-core" % V.pulsar4s,
       "com.sksamuel.pulsar4s" %% "pulsar4s-akka-streams" % V.pulsar4s,
+      "com.sksamuel.pulsar4s" %% "pulsar4s-circe" % V.pulsar4s,
```
Comment (amsjavan) [inline project/Dependencies.scala]: Removes extra deps

File: project/plugins.sbt [hunk 7]
```diff
@@ -1,11 +1,13 @@
 resolvers += "Flyway" at "https://davidmweber.github.io/flyway-sbt.repo"
 
+// todo: remove this after akka grpc power api version
```
Comment (amsjavan) [inline project/plugins.sbt]: Put link of PR

**Pull request merged**